Client/src/Components/DatasetUpload/JSONdatasetDownloadButton.tsx

```diff
@@ -0,0 +1,166 @@
+            name.push(
+                {
+                    "composition ": datasets.material[i].composition,
+                    "details": datasets.material[i].details
+                }
+            )
+        } return name
+    }
+    const authorArray = (): any => {
+        var name = []
+        for (var i in datasets.reference.authors) {
+            if (datasets.reference.authors[i].middleName) {
+                name.push(
+                    {
+                        "firstname ": datasets.reference.authors[i].firstName,
+                        "middlename": datasets.reference.authors[i].middleName,
+                        "lastname ": datasets.reference.authors[i].lastName
+                    }
+                )
+            }
+            else {
+                name.push(
+                    {
+                        "firstname ": datasets.reference.authors[i].firstName,
+                        "lastname ": datasets.reference.authors[i].lastName
+                    }
+                )
+            }
+
+        }
+        return name
+    }
+    const dataContentArray = (): any => {
+        var name = []
+        for (var i in datasets.data.contents) {
+            if (datasets.data.dataPointComments)
+                name.push(
+                    {
+                        "point": datasets.data.contents[i].point,
+                        "comment": datasets.data.dataPointComments[i]
+                    }
+                )
+            else
+                name.push(
+                    {
+                        "point": datasets.data.contents[i].point
+                    }
+                )
+
+        }
+        return name
+    }
+    let cgName: string
+    let scgName: string
+    let cat
+    const categoryName = () => {
+        categories.map(category => {
+            if (category.id == datasets.category) {
+                cgName = category.name
+            }
+        })
+        return cgName
+    }
+
+    const subcategoryName = () => {
+        categories.map(category => {
+            if (category.id == datasets.category) {
+                cat = category
+                cat.subcategories.map(subcategory => {
+                    if (subcategory.id == datasets.subcategory) {
+                        scgName = subcategory.name
+                    }
+                })
+            }
+        })
+        return scgName
+    }
+    const downloadedJSONDataDisplayed =
+    {
+        "Dataset name ": datasets.dataset_name,
+        "Material": materialArray(),
+        "Data type": datasets.data_type,
+        "Category": categoryName(),
+        "Subcategory ": subcategoryName(),
+        "reference": {
+            "Authors ": [authorArray()],
+            "Title ": datasets.reference.title,
+            "Type": datasets.reference.type,
+            "Publisher": datasets.reference.publisher,
+            "Volume, ": datasets.reference.volume,
+            "Pages, ": datasets.reference.pages,
```

Comment:
Would be nice to have null handling here, but not feasible given time constraints